### PR TITLE
fix(PVS/V1019): compound assignment expression is used inside condition

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: src/nvim/ops.c [hunk 1]
```diff
@@ -3893,7 +3893,8 @@ void ex_display(exarg_T *eap)
             msg_puts_attr("^J", attr);
             n -= 2;
           }
-          for (p = yb->y_array[j]; *p && (n -= ptr2cells(p)) >= 0; p++) {
+          for (p = yb->y_array[j]; *p && n >= 0; p++) {
+            n -= ptr2cells(p);
```
Comment (bfredl) [inline src/nvim/ops.c:3897]: doesn't this change the meaning of the code?

Comment (dundargoc) [reply src/nvim/ops.c:3897]: Ughhhh, you're right. I did make a small test to see if it was correct but it was a bad test it turns out. Thanks for the catch.

Comment (dundargoc) [reply src/nvim/ops.c:3897]: That PVS warning was for me :)

**Comment by dundargoc** [comment]
Bram wasn't a fan of my solution :(

I've reverted my change and opted to ignore the warning instead. The warning isn't *wrong* per se, but it's a bit more subjective than other rules.

**Comment by bfredl** [comment]
sounds good. squash?

**Pull request merged**